src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -277,7 +277,7 @@ public ProjectContext Build()
 
                     if (platformDependency != null)
                     {
-                        platformLibrary = libraries[new LibraryKey(platformDependency.Value.Name)];
+                        libraries.TryGetValue(new LibraryKey(platformDependency.Value.Name), out platformLibrary);
```

Comment:
I think ALL things that try to use a ProjectContext should verify that the ProjectContext is valid before using it, like so:

https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.Cli.Utils/Extensions/ProjectContextCollectionExtensions.cs#L14

If a ProjectContext has diagnostic errors, it should fail and those errors should be written to the screen.